src/components/products/product-card.tsx

```diff
@@ -1,14 +1,19 @@
-import { Heart } from '../assets';
-import { Product } from '../types';
+import { Product } from '../../types';
+import { HeartFill, HeartOutline } from '../../assets';
+import { useFavorites } from '../../context';
+import { useMemo } from 'react';
```

Comment:
Consider setting up [typescript barrel generator](https://marketplace.visualstudio.com/items?itemName=eliostruyf.vscode-typescript-exportallmodules) and [path aliases](https://blog.logrocket.com/using-path-aliases-cleaner-react-typescript-imports/) for the project